certbot/main.py

```diff
@@ -85,6 +85,18 @@ def _auth_from_available(le_client, config, domains=None, certname=None, lineage
     if lineage is None:
         # This will find a relevant matching lineage that exists
         action, lineage = _find_lineage_for_domains_and_certname(config, domains, certname)
+        if action in ("renew", "reinstall"):
+            # Found existing lineage; update config based on its configuration.
+            try:
+                # Some of the steps of reconstitute(), but without copying
```

Comment:
Aesthetically I now wonder if maybe we shouldn't have added a flag `include_domains=True` to `_reconstitute()` and just called that.  But I'm fine either way, I think!